### ISSUE TITLE
Distributed language model gets worse results than training on a single GPU

### ISSUE THREAD

**Issue by yeldarby** [fastai/fastai#1933]
**Describe the bug**
[I discovered](https://forums.fast.ai/t/should-training-distributed-with-bs-x-gpus-be-equivalent/42685/5) that training a language model using `to_distributed` on multiple GPUs gives worse results than with a single GPU.

I went ahead and tested this on another dataset to make sure it wasn't a quirk with my specific data and was able to replicate the problem on the `IMDB` dataset.

I've tracked down the problem to the way `torch.utils.data.distributed.DistributedSampler` is splitting up the data. I'm preparing a pull request that fixes this issue.

**Provide your installation details**
```text
=== Software === 
python        : 3.7.1
fastai        : 1.0.51
fastprogress  : 0.1.20
torch         : 1.0.0
nvidia driver : 410.66
torch cuda    : 10.0.130 / is available
torch cudnn   : 7401 / is enabled

=== Hardware === 
nvidia gpus   : 8
torch devices : 8
  - gpu0      : 8119MB | GeForce GTX 1070
  - gpu1      : 8119MB | GeForce GTX 1070
  - gpu2      : 8119MB | GeForce GTX 1070
  - gpu3      : 8119MB | GeForce GTX 1070
  - gpu4      : 8119MB | GeForce GTX 1070
  - gpu5      : 8119MB | GeForce GTX 1070
  - gpu6      : 8119MB | GeForce GTX 1070
  - gpu7      : 8119MB | GeForce GTX 1070

=== Environment === 
platform      : Linux-4.15.0-46-generic-x86_64-with-debian-stretch-sid
distro        : #49~16.04.1-Ubuntu SMP Tue Feb 12 17:45:24 UTC 2019
conda env     : Unknown
python        : /opt/conda/bin/python
sys.path      : /workspace/course-v3/nbs/dl1
/opt/conda/lib/python37.zip
/opt/conda/lib/python3.7
/opt/conda/lib/python3.7/lib-dynload

/opt/conda/lib/python3.7/site-packages
/opt/conda/lib/python3.7/site-packages/IPython/extensions
/root/.ipython
```

**To Reproduce**
I've [created a gist here](https://gist.github.com/yeldarby/07379fa373161282a6cf16b00c7f81d1) with the code I used for testing distributed language model training on the IMDB dataset.

**Expected behavior**
The baseline accuracy and loss from [`course-v3/lesson3-imdb.ipynb`](https://github.com/fastai/course-v3/blob/master/nbs/dl1/lesson3-imdb.ipynb) is as follows:

![imdb-baseline](https://user-images.githubusercontent.com/870796/55661277-f36bee00-57d8-11e9-9cab-edc87aff8a16.png)

**Actual behavior**
Training on 8 GPUs using [the code from the gist](https://gist.github.com/yeldarby/07379fa373161282a6cf16b00c7f81d1) shows much worse results:

![distributed-before](https://user-images.githubusercontent.com/870796/55662140-01bd0880-57df-11e9-8241-896e4c536442.png)

**Additional context**
I'm preparing a pull request with a proposed fix.
